autopush/endpoint.py

```diff
@@ -254,6 +273,49 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _extract_jwt(self, token, crypto_key):
+        # first split and convert the jwt.
+        sig = None
+        payload = {}
+
+        def fix_padding(string):
+            return string + '===='[len(sig) % 4:]
+
+        try:
+            (ehead, epayload, sig) = token.split('.', 2)
+            sig = fix_padding(sig)
```

Comment:
This changes the behavior of the next call to fix_padding, since fix_padding appends = based on the length of sig, and this call changes the length of sig. Later calls will always fail to append anything since sig is now right-padded.